homeassistant/components/sensor/demo.py

```diff
@@ -4,16 +4,68 @@
 For more details about this platform, please refer to the documentation
 https://home-assistant.io/components/demo/
 """
+import datetime as datetime
+import math
+import random
+import logging
+
+import voluptuous as vol
 from homeassistant.const import ATTR_BATTERY_LEVEL, TEMP_CELSIUS
 from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_NAME
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = datetime.timedelta(seconds=1)
+ICON = 'mdi:chart-line'
+
+CONF_UNIT = 'unit'
+CONF_AMP = 'amplitude'
+CONF_MEAN = 'mean'
+CONF_PERIOD = 'period'
+CONF_PHASE = 'phase'
+CONF_FWHM = 'spread'
+CONF_SEED = 'seed'
+
+DEFAULT_NAME = 'simulated'
+DEFAULT_UNIT = 'value'
+DEFAULT_AMP = 1
+DEFAULT_MEAN = 0
+DEFAULT_PERIOD = 60
+DEFAULT_PHASE = 0
+DEFAULT_FWHM = 0
+DEFAULT_SEED = 999
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_UNIT, default=DEFAULT_UNIT): cv.string,
+    vol.Optional(CONF_AMP, default=DEFAULT_AMP): vol.Coerce(float),
+    vol.Optional(CONF_MEAN, default=DEFAULT_MEAN): vol.Coerce(float),
+    vol.Optional(CONF_PERIOD, default=DEFAULT_PERIOD): cv.positive_int,
+    vol.Optional(CONF_PHASE, default=DEFAULT_PHASE): vol.Coerce(float),
+    vol.Optional(CONF_FWHM, default=DEFAULT_FWHM): vol.Coerce(float),
+    vol.Optional(CONF_SEED, default=DEFAULT_SEED): cv.positive_int,
+})
 
 
 # pylint: disable=unused-argument
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Demo sensors."""
+    name = config.get(CONF_NAME)
+    unit = config.get(CONF_UNIT)
+    amp = config.get(CONF_AMP)
+    mean = config.get(CONF_MEAN)
+    period = config.get(CONF_PERIOD)
+    phase = config.get(CONF_PHASE)
+    fwhm = config.get(CONF_FWHM)
+    seed = config.get(CONF_SEED)
+    random.seed(seed)
+
     add_devices([
         DemoSensor('Outside Temperature', 15.6, TEMP_CELSIUS, 12),
         DemoSensor('Outside Humidity', 54, '%', None),
```

Comment:
Yes I think we should drop these IMHO. Just make sure to update any tests that rely on the demo sensor platform.

Reply:
On the other hand, it's also nice to just need to put

```yaml
sensor:
  - platform: demo
```

in your config and have some demo sensors appear without any additional setup. I'm a bit divided on this one :/

Reply:
DemoSensor is just a SimulatedSensor with constant value, so could just swap the classes. We could just have a single demo sensor created with default values with no config set

Reply:
Sure! That's also the use case I thought of. I think one possibility would be to just update the previous `DemoSensor('Outside Humidity', 54, '%', None),` to use our new random data stream.

> It would be nice to have a live graph on the demo page?

AFAIK the demo page doesn't have a hass installation running somewhere and is purely front-end code. Therefore any change in this file wouldn't show up in the demo page - you'd have the front-end demo code for that instead.

Reply:
OK the default demo sensor will be identical to the current humidity demo sensor:
```yaml
DEFAULT_NAME = 'Outside Humidity'
DEFAULT_UNIT = '%'
DEFAULT_AMP = 0
DEFAULT_MEAN = 54
DEFAULT_PERIOD = 0
DEFAULT_PHASE = 0
DEFAULT_FWHM = 0
DEFAULT_SEED = 999
```